types/store/plugins.ts

```diff
@@ -31,20 +32,20 @@ export type PluginsState = {
 export type PluginComponent = {
     id: string;
     pluginId: string;
-    component: React.Component;
+    component: React.ComponentType<any>;
 };
 
 export type PostPluginComponent = {
     id: string;
     pluginId: string;
     type: string;
-    component: React.Component;
+    component: React.ComponentType<{post: Post}>;
 };
 
 export type AdminConsolePluginComponent = {
     pluginId: string;
     key: string;
-    component: React.Component;
+    component: React.ComponentType<any>;
```

Comment:
Is this really `any` component or is it a certain subset of components? We try to avoid `any` so it'd be nice to type this.

Reply:
I tried to go through it's usages, but I couldn't find anything that would suggest it's shape. I could replace `any` with `{}` which would postpone this problem to whenever some other migration from JS to TS that uses it would start complaining. Running `npm run check-types` with `{}` instead of `any` reports no new errors. What are your thoughts?

Reply:
Both are kinda the same thing, since they basically just short-circuit type checking by allowing basically anything. Usually the type checker should throw an error if you type it wrong, so maybe that could suggest it's shape? I'm 2/5 on this, if it's too much effort I'm okay with a comment explaining why the `any` or `{}` was used.

Reply:
Wouldn't `React.ComponentType<{}>` allow only for components without any properties? For example, when I remove `{post: Post}` from above, I get an error here: [components/rhs_card/rhs_card.tsx:120](https://github.com/mattermost/mattermost-webapp/blob/ffa369cc3068fce6c7cda4336e66e9e5d0b12957/components/rhs_card/rhs_card.tsx#L120).

With `{}`, instantiating a component with any props would raise an error and would require an update to the shape here.

I will try to dig deeper to find a shape, I will let you know how it goes